src/tpm2_wrap.c

```diff
@@ -1578,11 +1614,16 @@ int wolfTPM2_LoadEccPrivateKey(WOLFTPM2_DEV* dev, const WOLFTPM2_KEY* parentKey,
     rc = wolfTPM2_ImportEccPrivateKey(dev, parentKey, &keyBlob, curveId,
         eccPubX, eccPubXSz, eccPubY, eccPubYSz, eccPriv, eccPrivSz);
     if (rc == 0) {
-        rc = wolfTPM2_LoadKey(dev, &keyBlob, (WOLFTPM2_HANDLE*)&parentKey->handle);
+        rc = wolfTPM2_LoadKey(dev, &keyBlob,
+            (WOLFTPM2_HANDLE*)&parentKey->handle);
     }
 
     /* return loaded key */
-    XMEMCPY(key, &keyBlob, sizeof(WOLFTPM2_KEY));
+    key->handle.hndl = keyBlob.handle.hndl;
+    wolfTPM2_CopyAuth(&key->handle.auth, &keyBlob.handle.auth);
+    wolfTPM2_CopyName(&key->handle.name, &keyBlob.handle.name);
+    wolfTPM2_CopySymmetric(&key->handle.symmetric, &keyBlob.handle.symmetric);
+    wolfTPM2_CopyPub(&key->pub, &keyBlob.pub);
```

Comment:
In favor of readability, does it make sense to have a helper that combines these for copying WOLFTPM2_KEY , KEYBLOB and HANDLE structures?

This way it will again be a single line action and perhaps easier to follow

Reply:
Sure will cleanup.